pkgs/applications/networking/p2p/mldonkey/default.nix

```diff
@@ -2,52 +2,61 @@
   lib,
   stdenv,
   fetchurl,
-  fetchpatch,
+  autoreconfHook,
+  autoconf-archive,
   ocamlPackages,
+  pkg-config,
   zlib,
 }:
 
 stdenv.mkDerivation rec {
   pname = "mldonkey";
-  version = "3.1.7-2";
+  version = "3.2.1";
 
   src = fetchurl {
     url = "https://ygrek.org/p/release/mldonkey/mldonkey-${version}.tar.bz2";
-    sha256 = "b926e7aa3de4b4525af73c88f1724d576b4add56ef070f025941dd51cb24a794";
+    hash = "sha256-nAvjl6shkyVvUve6+vIQvynJO7R1i7GBj0YzdjuZa8c=";
```

Comment:
```nix
  src = fetchFromGitHub {
    owner = "ygrek";
    repo = "mldonkey";
    tag = "release-${lib.replaceStrings ["."] ["-"] version}";
    hash = "sha256-Dbb7163CdqHY7/FJY2yWBFRudT+hTFT6fO4sFgt6C/A=";
  };
```

It could be built from the sources directly since we are forced to generate the Makefile anyway.